runtime/src/main/scala/akka/grpc/internal/ChannelUtils.scala

```diff
@@ -52,32 +52,35 @@ object ChannelUtils {
       done: Promise[Done],
       channel: ManagedChannel,
       maxConnectionAttempts: Option[Int]): Unit = {
-    def monitor(previousState: ConnectivityState, connectionAttempts: Int): Unit =
-      if (maxConnectionAttempts.contains(connectionAttempts)) {
-        // shutdown is idempotent in ManagedChannelImpl
-        channel.shutdown()
-        done.tryFailure(new ClientConnectionException(s"Unable to establish connection after [$maxConnectionAttempts]"))
-      } else {
-        val currentState = channel.getState(false)
-        if (currentState == ConnectivityState.SHUTDOWN) {
+    def monitor(currentState: ConnectivityState, connectionAttempts: Int): Unit = {
+      //TODO: remove before merge
+      println(s"monitoring with state $currentState and connectionAttempts $connectionAttempts")
```

Comment:
Remove this? :)

Reply:
> I want to test and debug new monitoring before.

Cool, I'll wait with merging then - looking forward to it!

Reply:
@raboof Good news. Now netty heartbeat triggers `TRANSIENT_FAILURE`! Regarding `println` I think that it good idea to have such logging. What do you think about it and what kind of logger should I use?

Reply:
@raboof Is there any ETA of 0.7.4 release?

Reply:
We're aiming for 1.0.0 now.

Reply:
> I think that it good idea to have such logging.

That makes sense. For now I think it's OK to pass the `LoggingAdapter` available in `ClientState` on to here via the `NettyClientUtils`.

> Is there any ETA of 0.7.4 release?

We don't have one planned specifically, but I think we can release one soon. Perhaps we should also try and get #811 tested and included - have you seen that one? would you be interested in testing it?

Reply:
@raboof I think I can test #811 . Config are same and I just need to build project with that branch, am I right?

Reply:
Yes - you'll need to set `akka.grpc.client."*".grpc-load-balancing = "round_robin"`

Reply:
#811 has now been merged and a snapshot was released as version 0.7.3+72-96e15516 - do you think you'll have time to look into testing that?